File: Ejercicio 16/ejercicio16.java
import java.util.ArrayList;
import java.util.List;

class DetallePedido {
    private String producto;
    private int cantidad;

    public DetallePedido(String producto, int cantidad) {
        this.producto = producto;
        this.cantidad = cantidad;
    }

    public String getInfo() {
        return cantidad + "x " + producto;
    }
}

class Pedido {
    private int idPedido;
    private List<DetallePedido> detalles;

    public Pedido(int id) {
        this.idPedido = id;
        this.detalles = new ArrayList<>();
    }

    // Composición: El Pedido crea el Detalle internamente
    public void agregarItem(String nombre, int cant) {
        this.detalles.add(new DetallePedido(nombre, cant));
    }

    public void mostrarPedido() {
        System.out.println("ID Pedido: " + idPedido);
        for (DetallePedido d : detalles) {
            System.out.println(" -> " + d.getInfo());
        }
    }
}

public class Main {
    public static void main(String[] args) {
        Pedido miPedido = new Pedido(778);
        
        // No pasamos objetos creados, pasamos datos para que el pedido los cree
        miPedido.agregarItem("Pizza Familiar", 2);
        miPedido.agregarItem("Refresco 2L", 1);

        miPedido.mostrarPedido();
    }
}

File: Ejercicio 16/ejercicio16.py
class DetallePedido:
    def __init__(self, producto, cantidad):
        self.producto = producto
        self.cantidad = cantidad

class Pedido:
    def __init__(self, id_pedido):
        self.id_pedido = id_pedido
        self.detalles = []

    
    def agregar_producto(self, nombre, cantidad):
        nuevo_detalle = DetallePedido(nombre, cantidad)
        self.detalles.append(nuevo_detalle)


if __name__ == "__main__":
    pedido_app = Pedido(901)
    
    pedido_app.agregar_producto("Hamburguesa Doble", 3)
    pedido_app.agregar_producto("Papas Fritas", 2)

    print(f"Pedido #: {pedido_app.id_pedido}")
    for item in pedido_app.detalles:
        print(f"  - {item.cantidad} unidad(es) de {item.producto}")
